python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py

```diff
@@ -327,8 +318,20 @@ def is_ready(self, params: PipesParams) -> bool:
     # job has finished.
     def _get_log_path(self, params: PipesParams) -> Optional[str]:
         if self.log_path is None:
-            log_root_path = os.path.join(params["cluster_log_root"])
-            child_dirs = list(self.dbfs_client.list(log_root_path))
-            if len(child_dirs) > 0:
-                self.log_path = f"dbfs:{child_dirs[0].path}/driver/{self.remote_log_name}"
+            cluster_driver_log_root = params["extras"]["cluster_driver_log_root"]
+            try:
+                child_dirs = list(self.dbfs_client.list(cluster_driver_log_root))
+            except IOError:
+                child_dirs = []  # log root doesn't exist yet
+            match = next(
+                (
+                    child_dir
+                    for child_dir in child_dirs
+                    if child_dir.path and child_dir.path.endswith(self.remote_log_name)
+                ),
+                None,
+            )
+            if match:
+                self.log_path = f"dbfs:{match.path}"
```

Comment:
should it be a hard error if there is no match? What does that state represent?

Reply:
I don't think it should be a hard error because different log files can come into existence at different times. No match simply means the log file doesn't exist yet, i.e. `self.log_path` will remain `None`.